src/Dialect/ONNX/ONNXOps.hpp

```diff
@@ -24,73 +24,20 @@
 #include "mlir/IR/OpDefinition.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 
+#include "src/Dialect/ONNX/ONNXOpsDialect.hpp.inc"
 #include "src/Interface/HasOnnxSubgraphOpInterface.hpp"
 #include "src/Interface/ResultTypeInferenceOpInterface.hpp"
 #include "src/Interface/ShapeInferenceOpInterface.hpp"
+#define GET_TYPEDEF_CLASSES
+#include "src/Dialect/ONNX/ONNXOpsTypes.hpp.inc"
 
 namespace mlir {
 
 // OpSet level supported by onnx-mlir
 static constexpr int CURRENT_ONNX_OPSET = 13;
-
-class ONNXOpsDialect : public Dialect {
-public:
-  ONNXOpsDialect(MLIRContext *context);
-
-  /// Parse an instance of a type registered to the onnx dialect.
-  mlir::Type parseType(mlir::DialectAsmParser &parser) const override;
-
-  /// Print an instance of a type registered to the onnx dialect.
-  void printType(
-      mlir::Type type, mlir::DialectAsmPrinter &printer) const override;
-
-  /// Provide a utility accessor to the dialect namespace. This is used by
-  /// several utilities for casting between dialects.
-  static StringRef getDialectNamespace() { return "onnx"; }
-};
 } // end namespace mlir
 
 /// Include the auto-generated header file containing the declarations of the
 /// ONNX operations.
 #define GET_OP_CLASSES
 #include "src/Dialect/ONNX/ONNXOps.hpp.inc"
-
-namespace mlir {
```

Comment:
I think, for consistency with other MLIR dialects (LLVM dialect, SPRIV dialect, ...) the ONNX MLIR types should be in a header file (ONNXTypes.h).  I see that in those dialect they naming convention is to nest types in the dialect namespace, so alternatives are:
```
mlir::onnxmlir::ONNXStringType, or
mlir::onnx::ONNXStringType  <--- prob. more consistent
```

Reply:
With separate namespace, we will use mlir::onnx::StringType. The current ONNXStringType is just a hack to make the code compilable.

Reply:
ok. Whether we end up calling it  `mlir::onnx::ONNXStringType` or `mlir::onnx::ONNXStringType` doesn't make much of a difference. For now what you have is a good first step.